libs/tup-components/src/mfa/MfaQRPanel.tsx

```diff
@@ -1,68 +1,96 @@
 
 const MfaQRPanel: React.FC = () => {
   const { mutate, isLoading, data, isError } = useMfaPairTotp();
   return (
-    <div>
-      <div>
-        1. Open an approved{' '}
-        <a
-          href="https://docs.tacc.utexas.edu/basics/mfa/#mfaapps"
-          target="_blank"
-          rel="noreferrer"
+    <>
+      <form className={styles['mfa-form']}>
+        <FieldWrapper
+          name="mfa-phone-number"
+          label={
+            <>
+              Open an approved{' '}
+              <a
+                href="https://docs.tacc.utexas.edu/basics/mfa/#mfaapps"
+                target="_blank"
+                rel="noreferrer"
+              >
+                MFA pairing app
+              </a>{' '}
+              and scan the following QR code:
+            </>
+          }
+          error={
+            !data &&
+            isError && (
+              <>
+                Unable to display QR code. If this error persists,{' '}
+                <TicketCreateModal display="link">
+                  submit a ticket
+                </TicketCreateModal>
+                .
+              </>
+            )
+          }
         >
-          MFA pairing app
-        </a>{' '}
-        and scan the following QR code:
-      </div>
-      <div className={styles['qr-code-box']}>
-        <div>
-          {!data && !isLoading && (
-            <Button
-              className={styles['qr-button']}
-              onClick={() => mutate(null)}
-            >
-              Click to Generate QR Code
-            </Button>
-          )}
-          {!data && isLoading && <LoadingSpinner />}
-          {data && (
-            <img src={data.googleurl.img} alt="Google MFA QR Code"></img>
-          )}
-        </div>
-      </div>
-      {data && (
-        <div>
-          If your mobile device does not have a camera capable of scanning QR
-          codes, the following text code can be used:
-          <br />
-          <code>{data.otpkey.value_b32}</code> &nbsp;(
-          <Button
-            type="link"
-            onClick={() => navigator.clipboard.writeText(data.otpkey.value_b32)}
+          <div className={styles['qr-code-box']}>
+            {!data && !isLoading && (
+              <Button
+                id=""
+                className={styles['qr-button']}
+                onClick={() => mutate(null)}
+              >
+                Click to Generate QR Code
+              </Button>
+            )}
+            {!data && isLoading && <LoadingSpinner />}
+            {data && (
+              <img src={data.googleurl.img} alt="Google MFA QR Code"></img>
+            )}
+          </div>
+        </FieldWrapper>
+      </form>
+      {data && data.otpkey && (
+        <p className={styles['mfa-message']}>
+          Can't scan QR code?{' '}
+          <TextCopyModal
+            display="link"
+            title="Alternative Verification Code"
+            text={data.otpkey.value_b32}
+            hint={
```

Comment:
Can these hints/error messages be defined elsewhere? The markup become somewhat illegible when there are large blocks of JSX mixed in with the props.

Reply:
**Yes.** Moved. _Is 962d715 the simplest way to move it out and use it?_